bitbots_vision/src/bitbots_vision/vision_modules/ros_utils.py

```diff
@@ -0,0 +1,536 @@
+        Builds a balls message out of a list of ball messages
+        :param header: ros header of the new message. Mostly the header of the image
+        :param balls: A list of BallInImage messages
+        :return: balls msg
+        """
+        # create ball msg
+        balls_msg = BallsInImage()
+        # Set header
+        balls_msg.header.frame_id = header.frame_id
+        balls_msg.header.stamp = header.stamp
+        # Add balls
+        for ball in balls:
+            balls_msg.candidates.append(ball)
+        return balls_msg
+
+    @staticmethod
+    def build_ball_msg(top_ball_candidate):
+        """
+        Builds a ball message
+        :param top_ball_candidate: best rated ball candidate
+        :return: ball msg
+        """
+        # Create a empty ball message
+        ball_msg = BallInImage()
+        ball_msg.center.x = top_ball_candidate.get_center_x()
+        ball_msg.center.y = top_ball_candidate.get_center_y()
+        ball_msg.diameter = top_ball_candidate.get_diameter()
+        ball_msg.confidence = top_ball_candidate.get_rating()
+        return ball_msg
+
+    @staticmethod
+    def build_obstacles_msg(header, obstacles):
+        """
+        Builds a ObstaclesInImage message containing a list of obstacle messages
+        :param header: ros header of the new message. Mostly the header of the image
+        :param obstacles: a list of obstacle messages
+        :return: ObstaclesInImage message
+        """
+        # Create obstacle msg
+        obstacles_msg = ObstaclesInImage()
+        # Add header
+        obstacles_msg.header.frame_id = header.frame_id
+        obstacles_msg.header.stamp = header.stamp
+        # Add red obstacles
+        obstacles_msg.obstacles = obstacles
+        return obstacles_msg
+
+    @staticmethod
+    def build_obstacle_msgs(obstacle_color, detections):
+        """
+        Builds a list of obstacles for a certain color
+        :param obstacle_color: color of the obstacles
+        :param detections: obstacle candidates
+        :return: list of obstacle msgs
+        """
+        message_list = []
+        for detected_obstacle in detections:
+            obstacle_msg = ObstacleInImage()
+            obstacle_msg.color = obstacle_color
+            obstacle_msg.top_left.x = detected_obstacle.get_upper_left_x()
+            obstacle_msg.top_left.y = detected_obstacle.get_upper_left_y()
+            obstacle_msg.height = int(detected_obstacle.get_height())
+            obstacle_msg.width = int(detected_obstacle.get_width())
+            obstacle_msg.confidence = detected_obstacle.get_rating()
+            obstacle_msg.playerNumber = 42
+            message_list.append(obstacle_msg)
+        return message_list
+
+    @staticmethod
+    def build_field_boundary_msg(header, field_boundary):
+        """
+        Builds a list of obstacles for a certain color
+        :param obstacle_color: color of the obstacles
+        :param detections: obstacle candidates
```

Comment:
This is not included in the parameters this function uses

Reply:
Oh copy paste....